File: 766.toeplitz-matrix.go
func isToeplitzMatrix(matrix [][]int) bool {
	raw := len(matrix)
	col := len(matrix[0])

	for i := 0; i < raw; i++ {
		for j := 0; j < col; j++ {
			x := i+1
			y := j+1
			if !(x >= raw || y >= col || matrix[i][j] == matrix[x][y]) {
				return false
			}
		}
	}

	return true
}

File: 766.toeplitz-matrix.cpp
// CreateTime: 2019-12-31 11:19:38
class Solution {
public:
    bool isToeplitzMatrix(vector<vector<int>>& matrix) {
        if (matrix.size() == 0 || matrix[0].size() == 0) {
            return true;
        }

        int row = matrix.size();
        int col = matrix[0].size();

        for (int x = 0; x < row-1; x++) {
            for (int y = 0; y < col; y++) {
                auto newX = x+1;
                auto newY = y+1;
                if (newX >= row || newY >= col) {
                    continue;
                }

                if (matrix[newX][newY] != matrix[x][y]) {
                    return false;
                }
            }
        }

        return true;
    }
};
